File: data-structures/stack/java/Stack.java
public class Stack {
  Node head;
  private int size;

  class Node {
    int data;
    Node next;

    Node(int data) {
      this.data = data;
      this.next = null;
    }
  }

  public void push(int data) {
    Node newNode = new Node(data);

    if (head == null) {
      head = newNode;
      size++;
      return;
    }

    newNode.next = head;
    head = newNode;
    size++;
  }

  public int pop() {
    if (head == null) {
      System.out.println("Stack is empty");
    }

    int value = head.data;
    head = head.next;
    size--;

    return value;
  }

  public int top() {
    return head.data;
  }

  public int size() {
    return size;
  }

  public boolean isEmpty() {
    return size == 0;
  }
}

File: data-structures/stack/go/stack.go
package main

import (
	"fmt"
)

type StackNode struct {
	data int
	next *StackNode
}

type Stack struct {
	head *StackNode
	size int
}

func (s *Stack) Push(data int) {
	newNode := &StackNode{data: data}
	newNode.next = s.head
	s.head = newNode
	s.size++
}

func (s *Stack) Pop() int {
	if s.head == nil {
		fmt.Println("Stack is empty")
		return -1
	}
	val := s.head.data
	s.head = s.head.next
	s.size--
	return val
}

func (s *Stack) Top() int {
	if s.head == nil {
		fmt.Println("Stack is empty")
		return -1
	}
	return s.head.data
}

func (s *Stack) Size() int {
	return s.size
}

func (s *Stack) IsEmpty() bool {
	return s.size == 0
}

func main() {
	stack := &Stack{}

	stack.Push(23)
	stack.Push(6)
	stack.Push(69)
	stack.Push(64)
	stack.Push(1)
	stack.Push(4)
	stack.Push(47)

	fmt.Println("Top:", stack.Top())
	fmt.Println("Pop:", stack.Pop())
	fmt.Println("Pop:", stack.Pop())
	fmt.Println("Size:", stack.Size())
	fmt.Println("Empty or Not:", stack.IsEmpty())
}
